examples/js/lines/LineSegments2.js

```diff
@@ -54,6 +54,56 @@ THREE.LineSegments2.prototype = Object.assign( Object.create( THREE.Mesh.prototy
 
 	}() ),
 
+	raycast: ( function () {
+
+		var ray = new THREE.Ray();
+		var inverseMatrix = new THREE.Matrix4();
+
+		return function raycast( raycaster, intersects ) {
+
+			var precision = raycaster.linePrecision; // todo: calculate proper precision
+
+			inverseMatrix.getInverse( this.matrixWorld );
+			ray.copy( raycaster.ray ).applyMatrix4( inverseMatrix );
+
+			const geometry = this.geometry;
+			var pointOnSegment = new THREE.Vector3();
```

Comment:
It's better to create these vectors once in the closure scope and not every time `raycast()` is called.

Reply:
Fixed in 96dc01126 .